### PR TITLE
Cleanup the text during the re-bind of the Decimal and Integer view holders.

### LINKED ISSUE #2401: There is a duplicated invalid decimal entry in another decimal field.
**Describe the bug**
There is a duplicated invalid decimal entry in another decimal field.

**Component**
SDC library

**To Reproduce**
Add multiple decimal items to a single questionnaire form. 
Keep adding invalid entries to the decimal items. 
Start scrolling to add an invalid decimal entry to another field. 
The decimal entries are duplicated.

**Expected behavior**
Invalid decimal entries should not be duplicated.

**Screenshots**

https://github.com/google/android-fhir/assets/86107848/f6c8378c-2c0e-46ab-97f3-f528515de988

**Smartphone (please complete the following information):**
 - Device: Pixel2a emulator
 - Android version: Android 12

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by aditya-07**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #2401

**Description**
Updated the code to
1. Set the draft answer correctly.
2. Clear the answers when possible rebind occurs ( no draft or answer is available to be set).

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactory.kt [hunk 1]
```diff
@@ -52,11 +52,15 @@ internal object EditTextDecimalViewHolderFactory :
 
         val decimalStringToDisplay = questionnaireItemViewItemDecimalAnswer ?: draftAnswer
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactory.kt:53]: remove this variable

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactoryTest.kt [hunk 2]
```diff
@@ -447,4 +447,38 @@ class EditTextDecimalViewHolderFactoryTest {
...
+    val questionnaireViewItem =
+      QuestionnaireViewItem(
+        Questionnaire.QuestionnaireItemComponent(),
+        QuestionnaireResponse.QuestionnaireResponseItemComponent(),
+        validationResult = NotValidated,
+        answersChangedCallback = { _, _, _, _ -> },
+      )
+    viewHolder.bind(questionnaireViewItem)
+    viewHolder.itemView.findViewById<TextInputEditText>(R.id.text_input_edit_text).apply {
+      setText("1.1.1.1")
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactoryTest.kt:462]: to make this test case more realistic, can we just add a draft answer in the questionnaire view item?

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextIntegerViewHolderFactoryTest.kt [hunk 3]
```diff
@@ -369,4 +369,38 @@ class EditTextIntegerViewHolderFactoryTest {
...
+    val questionnaireViewItem =
+      QuestionnaireViewItem(
+        Questionnaire.QuestionnaireItemComponent(),
+        QuestionnaireResponse.QuestionnaireResponseItemComponent(),
+        validationResult = NotValidated,
+        answersChangedCallback = { _, _, _, _ -> },
+      )
+    viewHolder.bind(questionnaireViewItem)
+    viewHolder.itemView.findViewById<TextInputEditText>(R.id.text_input_edit_text).apply {
+      setText("9999999999")
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextIntegerViewHolderFactoryTest.kt:384]: ditto

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactoryTest.kt (hunk 2 above)
Comment (aditya-07) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactoryTest.kt]: Done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextIntegerViewHolderFactoryTest.kt (hunk 3 above)
Comment (aditya-07) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/factories/EditTextIntegerViewHolderFactoryTest.kt]: done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactory.kt (hunk 1 above)
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/factories/EditTextDecimalViewHolderFactory.kt]: Done

**Pull request merged**